sdk/keyvault/Azure.Security.KeyVault.Administration/src/RestoreOperationInternal.cs

```diff
@@ -100,18 +87,47 @@ internal RestoreOperationInternal(RestoreDetailsInternal value, Response respons
             _client = client;
             _response = response;
             _value = value;
-            _id = value.JobId;
+            _id = value switch
+            {
+                SelectiveKeyRestoreDetailsInternal r => r.JobId,
+                RestoreDetailsInternal r => r.JobId,
+                null => default,
+                _ => throw new ArgumentException("Unknown type", nameof(value))
```

Comment:
Nit: these seem more like `InvalidOperationException` cases since the problem is with the `_value` data - nothing to do with property itself.